Automattic/wp-calypso
# Gutenboarding: Add feature flag to redirect to Site Editor upon Site Creation

#### Changes proposed in this Pull Request

Addressing item 1 from paYE8P-s6:

>1. Onboarding integration
>    - Expose a new test route in Horizon that will start the FSE tailored Gutenboarding
>    - Provide a different set of layouts/preview based on Seedlet Blocks theme
>    - At the end of the flow land users in the Site Editor

This PR watches out for a feature flag, and if it is set:
- Upon site creation, it sets the `site_creation_flow` option to `gutenboarding-site-editor` (rather than just `gutenboarding`)
- It takes the user to the site editor rather than the post (content) editor.

Fixes #43287.

#### Follow-up

Enabling feature flags by setting them via the `flags` qarg is currently not supported in the wpcalypso and horizon envs. I will file a PR to amend this.

#### Testing instructions

- ~Apply D44941-code to your WP.com sandbox, and sandbox the API~ _Edit: No longer needed_
- Go to `http://calypso.localhost:3000/new?flags=gutenboarding/site-editor`
- Go through the onboarding steps and create a new site.
- You should be taken to the full site editor. **Note that this currently does _not_ reflect the previously picked design -- that's for another iteration.** Instead, it should look somewhat like this (note the page and template selector in the top bar -- `Home / front-page` in the screenshot):

![image](https://user-images.githubusercontent.com/96308/84685080-ecdb1980-af39-11ea-8c21-7353f7933ecc.png)
